### ISSUE TITLE
Feature Request: Original shader source in Vulkan captures

### ISSUE THREAD

**Issue by jshofmann** [baldurk/renderdoc#1703]
## Description

It would be great when debugging a shader in a Vulkan capture if the original shader source was displayed instead of GLSL code generated by SPIRV-Cross in the shader source window displayed via the View link from the Pipeline State window. The code generated by SPIRV-Cross does not contain the comments from the original shader source code (which we use to trivially identify the source asset, among other purposes) and may not have a 1:1 correspondence to the original source, making it more difficult to place fixes in their proper locations in the source asset. 

## Environment

* RenderDoc version: 1.6
* Operating System: Windows 10 Pro build 18363
* Graphics API: Vulkan 1.1

**Comment by baldurk** [comment]
This isn't a RenderDoc issue. The shader viewer displays the original source when it's provided in the SPIR-V blob in an `OpSource`, you will need to configure your shader build pipeline to generate those `OpSource` entries.

**Comment by jshofmann** [comment]
Thanks, I had suspected this was more of a SPIRV issue but numerous web searches were inconclusive. For the purposes of anyone who finds this report, I did get it working by using glslang::TIntermediate::setSourceFile() and glslang::TIntermediate::addSourceText(). (I use glslang to compile the GLSL and convert it to SPIRV because shaderc did not exist when my toolchain was first written.) As far as I can tell even if I did use shaderc it does not call either function, it's up to the client to do that.
